public/main.js

```diff
@@ -96,26 +96,26 @@ Contact.prototype.onCancelClicked = function() {
 
 Contact.prototype.clearAddForm = function() {
 	this.addForm.find("input[type='text']").each(function() {
-		$(this).val("");
-		if($(this).attr("name").match(/\d$/)) {
-			$(this).siblings().each(function() {
-				$(this).remove()
-			});
-			$(this).parent().remove();
-			$(this).remove();
-		}
+	  $(this).val("");
+	  if($(this).attr("name").match(/\d$/)) {
+	    $(this).siblings().each(function() {
+	      $(this).remove()
+	    });
+	    $(this).parent().remove();
+	    $(this).remove();
```

Comment:
If you removed the parent on the previous line, $(this) should be removed along with it. This line _should_ be unnecessary